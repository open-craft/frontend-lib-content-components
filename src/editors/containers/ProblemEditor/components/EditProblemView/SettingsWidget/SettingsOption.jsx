import React, { useState } from 'react';
import { Collapsible, Icon, Card, } from '@edx/paragon';
import { KeyboardArrowUp, KeyboardArrowDown } from '@edx/paragon/icons';
import PropTypes from 'prop-types';

export const SettingsOption = ({
    title,
    summary,
    children
}) => {
    const [cardCollapsed, setCardCollapsed] = useState(false)

    return (
        <Card>
            <Card.Section>
                <Collapsible.Advanced
                    open={cardCollapsed}
                    onToggle={() => setCardCollapsed(!cardCollapsed)}
                >
                    <Collapsible.Trigger className="collapsible-trigger d-flex">
                        <span className="flex-grow-1">{title}</span>
                        <Collapsible.Visible whenClosed>
                            <Icon src={KeyboardArrowDown} />
                        </Collapsible.Visible>
                        <Collapsible.Visible whenOpen>
                            <Icon src={KeyboardArrowUp} />
                        </Collapsible.Visible>
                    </Collapsible.Trigger>
                </Collapsible.Advanced>
            </Card.Section>
            <Card.Section>
                <Collapsible.Advanced
                    open={!cardCollapsed}
                >
                    <Collapsible.Body className="collapsible-body">
                        <span>{summary}</span>
                    </Collapsible.Body>
                </Collapsible.Advanced>
                <Collapsible.Advanced
                    open={cardCollapsed}
                >
                    <Collapsible.Body className="collapsible-body">
                        {children}
                    </Collapsible.Body>
                </Collapsible.Advanced>
            </Card.Section>
        </Card>
    )
}

SettingsOption.propTypes = {
    title: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
};

export default SettingsOption;