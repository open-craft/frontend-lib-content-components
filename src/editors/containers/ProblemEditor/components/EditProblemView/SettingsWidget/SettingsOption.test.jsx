import React from 'react';
import { shallow } from 'enzyme';
import SettingsOption from './SettingsOption';

describe('SettingsOption', () => {
  describe('render', () => {
    const testContent = (<h1>My test content</h1>);
    test('snapshot: renders correct', () => {
      expect(shallow(<SettingsOption title="Settings Option Title" summary="Settings Option Summary">{testContent}</SettingsOption>)).toMatchSnapshot();
    });
  });
});
