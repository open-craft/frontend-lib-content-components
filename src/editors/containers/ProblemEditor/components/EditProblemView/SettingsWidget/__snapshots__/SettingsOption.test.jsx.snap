// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsOption render snapshot: renders correct 1`] = `
<Card>
  <Card.Section
    className="settingsCardTitleSection"
  >
    <Collapsible.Advanced
      onToggle={[Function]}
      open={false}
    >
      <Collapsible.Trigger
        className="collapsible-trigger d-flex"
      >
        <span
          className="flex-grow-1"
        >
          Settings Option Title
        </span>
        <Collapsible.Visible
          whenClosed={true}
        >
          <Icon />
        </Collapsible.Visible>
        <Collapsible.Visible
          whenOpen={true}
        >
          <Icon />
        </Collapsible.Visible>
      </Collapsible.Trigger>
    </Collapsible.Advanced>
  </Card.Section>
  <Card.Section>
    <Collapsible.Advanced
      open={true}
    >
      <Collapsible.Body
        className="collapsible-body"
      >
        <span>
          Settings Option Summary
        </span>
      </Collapsible.Body>
    </Collapsible.Advanced>
    <Collapsible.Advanced
      open={false}
    >
      <Collapsible.Body
        className="collapsible-body"
      >
        <h1>
          My test content
        </h1>
      </Collapsible.Body>
    </Collapsible.Advanced>
  </Card.Section>
</Card>
`;
